services/thermal_mgr/thermal_mgr.c

```diff
@@ -38,23 +42,47 @@ void initThermalSystemManager(lm75bd_config_t *config) {
   thermalMgrQueueHandle = xQueueCreateStatic(
     THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
     thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if (event->type == OS_HANDLER_EVENT){
+    xQueueSendToFront(thermalMgrQueueHandle, event, ( TickType_t ) 0);
+  }  
+  else if (event->type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD){
+    xQueueSend(thermalMgrQueueHandle, event, ( TickType_t ) 10 );
+  }
   return ERR_CODE_SUCCESS;
 }
 
+BOOL readTempState = FALSE;
```

Comment:
Is this global flag necessary? The task will know that the event came from osHandler if the type was OS_HANDLER_EVENT.